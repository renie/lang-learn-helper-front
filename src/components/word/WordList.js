import React from 'react'

import './wordList.css'

const WordList = ({words, WordListItemComponent}) => {
    return (
        <ul className="mansory">
            {(words && words.length)
                ? words.map((word, index) => <WordListItemComponent key={word._id}  word={word}/>)
                : <li>No words do be listed</li>}
        </ul>
    )
}

export default WordList
