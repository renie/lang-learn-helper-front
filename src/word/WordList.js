import React from 'react'

import WordListItem from './WordListItem'

const WordList = ({words}) => {
  return (
    <ul>
        {words.map((word, index) => <WordListItem key={word._id} word={word}/>)}
    </ul>
  )
}

export default WordList
