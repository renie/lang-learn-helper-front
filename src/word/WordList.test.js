import React from 'react'
import ReactDOM from 'react-dom'
import WordList from './WordList'
import renderer from 'react-test-renderer'


it('renders correctly empty', () => {
    const tree = renderer
                    .create(<WordList 
                                words={undefined}/>)
                    .toJSON()

    expect(tree).toMatchSnapshot()
})

it('renders correctly empty', () => {
    const tree = renderer
                    .create(<WordList 
                                words={[]}/>)
                    .toJSON()

    expect(tree).toMatchSnapshot()
})

it('renders correctly not empty', () => {
 
    const words = [{
            "name":"used",
            "meanings":[{
                "id": 1,
                "wordClass": "verb",
                "example": "He used my pen.",
                "description": "Verb use in past tense."
            },
            {
                "id":2,
                "wordClass":"adjective",
                "example":"My pen is used.",
                "description":"Means the object is not new anymore."
        }],
        "_id":"BcNdVZy038MWxqrg"
    }]

    const tree = renderer
                    .create(<WordList 
                                words={words}/>)
                    .toJSON()

  expect(tree).toMatchSnapshot()
})
