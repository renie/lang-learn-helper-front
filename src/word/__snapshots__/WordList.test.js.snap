// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly empty 1`] = `
<ul
  className="mansory"
>
  <li>
    No words do be listed
  </li>
</ul>
`;

exports[`renders correctly empty 2`] = `
<ul
  className="mansory"
>
  <li>
    No words do be listed
  </li>
</ul>
`;

exports[`renders correctly not empty 1`] = `
<ul
  className="mansory"
>
  <li
    className="card wordListItem"
  >
    <header
      className="card-header"
    >
      <p
        className="title"
      >
        used
      </p>
    </header>
    <div
      className="card-content"
    >
      <div
        className="content"
      >
        <p
          className="subtitle"
        >
          Meanings:
        </p>
        <ul>
          <li>
            <strong>
              verb
            </strong>
            <p>
              Verb use in past tense.
            </p>
            <em>
              Ex.: 
              He used my pen.
            </em>
          </li>
          <li>
            <strong>
              adjective
            </strong>
            <p>
              Means the object is not new anymore.
            </p>
            <em>
              Ex.: 
              My pen is used.
            </em>
          </li>
        </ul>
      </div>
    </div>
  </li>
</ul>
`;
